crates/merkle-tree/src/storage.rs

```diff
@@ -0,0 +1,164 @@
+use anyhow::Context;
+use bitvec::prelude::Msb0;
+use stark_hash::Felt;
+
+use crate::Node;
+
+/// Read-only storage used by the [Merkle tree](crate::tree::MerkleTree).
+pub trait Storage {
+    type Error: std::error::Error + Send + Sync + 'static;
+
+    fn get(&self, node: &Felt) -> Result<Option<Node>, Self::Error>;
+}
+
+/// Wrapper around [anyhow::Error].
+#[derive(Debug)]
+pub struct AnyhowError(anyhow::Error);
+impl std::fmt::Display for AnyhowError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!("{}", self.0))
+    }
+}
+impl std::error::Error for AnyhowError {}
+impl From<anyhow::Error> for AnyhowError {
+    fn from(value: anyhow::Error) -> Self {
+        AnyhowError(value)
+    }
+}
```

Comment:
`thiserror` ?

Reply:
Done with thiserror now. Wrapper is required because `anyhow::Error` is not actually a useable `Error` type. One could define `Storage::Error` type differently I suppose, but its difficult to compose with `anyhow::Error`.